### DISCUSSION TITLE
Must the loose property of class properties plugin be true when the legacy property of decorators plugin is true? and why?

### DISCUSSION THREAD

**Discussion by Hughe5** [babel/babel#14133] [Q&A] [answered]
```
plugins: [
    ['@babel/plugin-proposal-decorators', { legacy: true }],
    ['@babel/plugin-proposal-class-properties', { loose: true }],
  ],
```
When using decorators plugin and class properties plugin at the same time,
when the legacy property of the decorators plugin is set to true,
must the loose property of the class properties plugin be set to true?

> When using the legacy: true mode, the setPublicClassFields assumption must be enabled to support the @babel/plugin-proposal-decorators.

I don't understand this sentence, how class properties plugin support decorators plugin?

please.

**Comment by nicolo-ribaudo** [comment, marked as answer]
Yes, when you use legacy decorators you cannot use spec-compliant class properties because they are not compatible. You can either pass `loose: true`, or use the `setPublicClassFields` assumption:

```jsonc
plugins: [
    ['@babel/plugin-proposal-decorators', { legacy: true }],
    ['@babel/plugin-proposal-class-properties', { loose: true }],
],
```
```jsonc
assumptions: {
  setPublicClassFields: true
},
plugins: [
    ['@babel/plugin-proposal-decorators', { legacy: true }],
    '@babel/plugin-proposal-class-properties',
],
```

More info about assumptions: https://babeljs.io/docs/en/assumptions

FYI, we are working on the next version of our decorators plugin and it will be available in Babel 7.17.0: https://github.com/babel/babel/pull/14004

**Reply by Hughe5** [reply]
Thank you so much, your answer is very helpful to me.

 > when you use legacy decorators you cannot use spec-compliant class properties because they are not compatible.

Could you give me an example where it is not compatible?

Because in my application, babel config is:
```
plugins: [
    ['@babel/plugin-proposal-decorators', { legacy: true }],
    ['@babel/plugin-proposal-class-properties', { loose: false }],
  ],
```
I have no errors with my application.

**Reply by nicolo-ribaudo** [reply]
Oh, I am investigating and it looks like it's not needed anymore!

Here I'm testing the current Babel version, and Babel 7.0.0: https://runkit.com/nicolo-ribaudo/61ded52f332f120008922ae3
In Babel 7.0.0 it was necessary to use loose mode (you can see `initializerWarningHelper` in the Babel 7.0.0 output without loose mode, which means "something went wrong"), but it looks like it's not needed anymore :smile: It's probably thanks to https://github.com/babel/babel/pull/8130 + https://github.com/babel/babel/pull/9083, released in v7.2.0.

**Reply by Hughe5** [reply]
Nice!
Looking forward to your exact conclusion, and remember to update the doc please ❤️  🚀
